### PR TITLE
Add some System.Runtime.InteropServices tests

### PR DESCRIPTION
**Pull request by hughbe**
Cleanup some existing test code, add tests to increase the code coverage numbers in the library.

I'm planning on submitting a PR after this adding some missing coverage for things that reside in coreclr

### CODE REVIEW

**Comment by hughbe** [comment]
Updated this PR to be a bit smaller, will split out the test additions for classes missing any unit tests

File: src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/ComAwareEventInfo.cs [hunk 1]
```diff
@@ -114,13 +92,13 @@ private static void GetDataForComInvocation(System.Reflection.EventInfo eventInf
             if (comEventInterfaces.Length > 1)
             {
                 // TODO: event strings need to be localizable
```
Comment (danmosemsft) [inline src/System.Runtime.InteropServices/src/System/Runtime/InteropServices/ComAwareEventInfo.cs]: if you feel like it you could put the strings in a resx.

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/AllowReversePInvokeCallsAttributeTests.cs [hunk 2]
```diff
@@ -11,18 +11,15 @@ namespace System.Runtime.InteropServices.Tests
...
         [Fact]
         public void Exists()
         {
-            var type = typeof(AllowReversePInvokeCallsAttributeTests);
-            var method = type.GetTypeInfo().DeclaredMethods.Single(m => m.Name == "Func");
-            var attr = method.GetCustomAttributes(typeof(AllowReversePInvokeCallsAttribute), false).OfType<AllowReversePInvokeCallsAttribute>().SingleOrDefault();
-            Assert.NotNull(attr);
+            Type type = typeof(AllowReversePInvokeCallsAttributeTests);
+            MethodInfo method = type.GetTypeInfo().DeclaredMethods.Single(m => m.Name == "Func");
+            AllowReversePInvokeCallsAttribute attribute = Assert.Single(method.GetCustomAttributes< AllowReversePInvokeCallsAttribute>(false));
```
Comment (danmosemsft) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/AllowReversePInvokeCallsAttributeTests.cs]: it's nice to name boolean parameters
` ... Assert.Single(method.GetCustomAttributes< AllowReversePInvokeCallsAttribute>(inherit: false));` ?

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/HandleCollectorTests.cs [hunk 3]
```diff
@@ -95,36 +123,27 @@ public static void CountOverflow()
...
             }
 
-            Tuple<int, int, int> postLowLimitState = new Tuple<int, int, int>(
-                GC.CollectionCount(0),
-                GC.CollectionCount(1),
-                GC.CollectionCount(2));
+            (int gen0, int gen1, int gen2) postLowLimitState = (GC.CollectionCount(0),GC.CollectionCount(1), GC.CollectionCount(2));
 
-            Assert.True(initialGcState.Item1 + initialGcState.Item2 + initialGcState.Item3 < postLowLimitState.Item1 + postLowLimitState.Item2 + postLowLimitState.Item3, "Low limit handle did not trigger a GC");
+            Assert.True(initialGcState.gen0 + initialGcState.gen1 + initialGcState.gen2 < postLowLimitState.gen0 + postLowLimitState.gen1 + postLowLimitState.gen2, "Low limit handle did not trigger a GC");
```
Comment (danmosemsft) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/HandleCollectorTests.cs]: nice use of value tuple

File: src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/RuntimeEnvironmentTests.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        public void GetRuntimeInterfaceAsIntPtr_Invoke_ThrowsPlatformNotSupportedException()
+        {
+            Assert.Throws<PlatformNotSupportedException>(() => RuntimeEnvironment.GetRuntimeInterfaceAsIntPtr(Guid.Empty, Guid.Empty));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "FromGlobalAccessCache always returns true on .NET Core.")]
+        public void FromGlobalAccessCache_INvoke_ReturnsFalse()
+        {
+            Assert.False(RuntimeEnvironment.FromGlobalAccessCache(typeof(RuntimeEnvironmentTests).Assembly));
```
Comment (danmosemsft) [inline src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/RuntimeEnvironmentTests.cs]: nit extra line

**Pull request merged**